src/operator/setup.rs

```diff
@@ -0,0 +1,185 @@
+    where
+        C: Clone + Debug + DeserializeOwned + Resource<DynamicType = ()> + Serialize,
+    {
+        self.client.delete(command);
+        Ok(())
+    }
+
+    /// Returns all available pods in the cluster via the name label.
+    pub fn get_current_pods(&self) -> Vec<Pod> {
+        let current_pods: ObjectList<Pod> = self.client.list_labeled(&format!(
+            "app.kubernetes.io/name={}",
+            &self.options.pod_name_label
+        ));
+        current_pods.items
+    }
+
+    /// Returns all available nodes in the cluster. Can be used to determine the expected pods
+    /// for tests (depending on the custom resource)
+    pub fn get_available_nodes(&self) -> Vec<Node> {
+        let available_nodes: ObjectList<Node> = self.client.list_labeled("");
+        available_nodes.items
+    }
+
+    /// A "busy" wait for all pods to be terminated and cleaned up.
+    pub fn wait_for_pods_terminated(&self) -> Result<()> {
+        let now = Instant::now();
+
+        while now.elapsed().as_secs() < self.timeouts.pods_terminated.as_secs() {
+            let pods = self.get_current_pods();
+
+            if pods.is_empty() {
+                return Ok(());
+            }
+
+            println!("Waiting for {} Pod(s) to terminate", pods.len());
+            thread::sleep(Duration::from_secs(1));
+        }
+
+        Err(anyhow!(
+            "Pods did not terminate within the specified timeout of {} second(s)",
+            self.timeouts.pods_terminated.as_secs()
+        ))
+    }
+}
+
+impl<T> TestCluster<T>
+where
+    T: Clone + Conditions + Debug + DeserializeOwned + Resource<DynamicType = ()> + Serialize,
+{
+    /// Creates or updates a custom resource and waits for the cluster to be up and running
+    /// within the provided timeout. Depending on the cluster definition we hand in the number
+    /// of created pods we expect manually.
+    pub fn create_or_update(&mut self, cluster: &T, expected_pod_count: usize) -> Result<()> {
+        self.apply(cluster)?;
+        self.wait_ready(expected_pod_count)?;
+        Ok(())
+    }
+
+    /// A "busy" (2 second sleep) wait for the cluster to be ready. We check the condition_type
+    /// and the expected pods that should be up and running.
+    pub fn wait_ready(&self, expected_pod_count: usize) -> Result<()> {
+        let now = Instant::now();
+
+        let name = self.cluster.as_ref().unwrap().name();
+
+        while now.elapsed().as_secs() < self.timeouts.cluster_ready.as_secs() {
+            println!("Waiting for [{}/{}] to be ready...", T::kind(&()), name);
+
+            let cluster: T = self.client.find_namespaced(&name).unwrap();
+
+            if let Some(conditions) = cluster.conditions() {
+                for condition in conditions {
+                    if condition.type_ == self.options.cluster_ready_condition_type
+                        // TODO: use operator-rs ConditionStatus?
+                        && condition.status == *"False"
```

Comment:
The comparison should work without the dereference operator:

```suggestion
                        && condition.status == "False"
```